vacabot/models/vacation.js

```diff
@@ -17,7 +17,15 @@ const vacationSchema = new Schema({
 });
 
 vacationSchema.methods.reduceVacationBalance = function () {
```

Comment:
The idea looks good. But loses the single responsibility principle.
This function gives a meaning that its job is to reduce vacations, and that's it.

Suggestion: You can check if it is valid and return true/false, and in `actionSubmitVacationRequest` you can notify the employee or the manager based on this boolean value.

Reply:
Hi @mukarramali, I have updated my changes as per you required, and thanks for your suggestion. I am not able to create a new pull request. So, please check this out.